### ISSUE TITLE
Plasma 6 compatibility?

### ISSUE THREAD

**Issue by mmcnutt** [Zren/breeze-alphablack#28]
I'm not sure if you're still maintaining the alphablack control widget, but the widget does not appear to work with Plasma 6. The themeing is still applied from before the upgrade, however the widget no longer shows up in the tray. 

The python script still appears to be effective for changing parameters tho

**Comment by D3SOX** [comment]
Sadly this seems to be unmaintained too https://github.com/Zren/plasma-applet-alphablackcontrol
I will try and see if I'm able to update the code to work with Plasma 6.
I found docs for it: https://develop.kde.org/docs/plasma/widget/porting_kf6/

**Comment by D3SOX** [comment]
![image](https://github.com/Zren/breeze-alphablack/assets/24937357/f93888f4-5385-4626-9ddb-3ba6216534ec)

I'm not that experienced with coding Plasmoids so it looks really buggy right now and you need to resize the applet as the scrolling is broken but at least the applet works now.

Changing colors and the opacity sliders work, the radio buttons for the style are a bit glitchy and the buttons to apply/reset title bar colors are missing

If anybody wants to help: https://github.com/D3SOX/plasma6-applet-alphablackcontrol

**Comment by dhruv8sh** [comment]
Okay so I have the applet running almost perfectly on my system. I need instructions on installing the theme itself so I can test and debug it.

![image](https://github.com/Zren/breeze-alphablack/assets/67322047/59e3016c-dc5e-4543-8419-141bc3a1fc4c)


Also I wrote a quick guide to plasma 6 applets. Here: https://medium.com/@dhruv8sh_34505/write-an-applet-for-plasma-6-0b8fd3a0334f

**Comment by D3SOX** [comment]
Works perfectly fine for me with your repo
![image](https://github.com/Zren/breeze-alphablack/assets/24937357/8d5188bb-baf9-439b-9944-dbc10a23916d)


I've had the theme installed before via the KDE Store, I think. The theme might also need a port from `metadata.desktop` to `metdata.json`

**Comment by D3SOX** [comment]
![image](https://github.com/Zren/breeze-alphablack/assets/24937357/558fd144-da1e-448f-957b-6ec68dc91943)
Ok radio buttons are also broken

And color pickers and also manually entering colors don't work. They worked in my repo. I think it's because you haven't migrated the `kuser.loginName` yet.

**Comment by dhruv8sh** [comment]
I am aware of this, I haven't pushed it yet. At least ask next time, lol.
I also ported the .desktop to .json.

**Comment by D3SOX** [comment]
Oh ok I didn't know just wanted to test for you sorry

**Comment by dhruv8sh** [comment]
Latest commit should work just fine. Test it out and let me know if it works.

**Comment by D3SOX** [comment]
Radio buttons work fine now.

Changing colors in the UI works but has no effect

And what's also an issue is that I can't see the applet by default in the systray and have to enable it every time
![image](https://github.com/Zren/breeze-alphablack/assets/24937357/7735535d-1d9c-4471-8690-5a57752f9c94)

**Comment by dhruv8sh** [comment]
This will take up too much time! I need to find a way to install the theme instead.
Edit: I have succeed in installing the theme. I will not debug it myself and notify you when the applet works completely.

**Comment by dhruv8sh** [comment]
Check now, seems to be working for me!

**Comment by D3SOX** [comment]
Yup changing colors works but I still think the effect should be automatically shown, dunno why it's not "relevant"

**Comment by dhruv8sh** [comment]
I don't get what you mean, elaborate please?

**Comment by D3SOX** [comment]
I think maybe `widgetsUnlocked` does not work correctly on Plasma 6 or some check when to show the applet

See previously you didn't have to manually add the Control Widget to use it. It would be inside the System tray (apparently when `widgetsUnlocked` not sure about that variable)

I think it was there in the overflow menu (bad visual representation):
![image](https://github.com/Zren/breeze-alphablack/assets/24937357/2b79c3e9-b432-4928-903c-057640f300f9)


Now it's not there when I want to use it I must set this to always shown

![image](https://github.com/Zren/breeze-alphablack/assets/24937357/7735535d-1d9c-4471-8690-5a57752f9c94)

Then it's directly an icon inside the systray

![image](https://github.com/Zren/breeze-alphablack/assets/24937357/e8a12594-7d5c-4747-9ce7-a5a4cdea59db)

I've also tried to manually add it to my panel but I only see that when I'm in edit mode so I can't click it

**Comment by dhruv8sh** [comment]
Fixed! Since this issue does not seem to be watched over, I will publish it on the store myself.

**Comment by D3SOX** [comment]
> Fixed! Since this issue does not seem to be watched over, I will publish it on the store myself.

Thank you so much! Yeah, I think that's a good idea, the repos seem to be unmaintained

I'm gonna create an AUR package for it

**Comment by D3SOX** [comment]
I just noticed the control applet only works when the theme is installed as user because of the way it works internally to change the settings, so packaging the theme with the current structure doesn't work.

Nevertheless I've packaged the control plasmoid: https://aur.archlinux.org/packages/plasma6-applets-breeze-alphablack

**Comment by D3SOX** [comment]
So I think it would be best if you could also publish the theme on the KDE Store with the control plasmoid as a dependency so it gets automatically installed, like it was before ^^

I also found another small bug: When setting it to be always hidden (as you are not really changing this that often) makes it so that when you click it in the overflow menu it doesn't open up.

![image](https://github.com/Zren/breeze-alphablack/assets/24937357/c90e20b0-1583-403e-8272-fe43c32025c4)

**Comment by dhruv8sh** [comment]
Please open an issue in the new repo. I will get to this tomorrow.

**Comment by D3SOX** [comment]
I would've posted it there but issues are not open in your fork

**Comment by dhruv8sh** [comment]
😅yes that's on me. They're open now.

**Comment by D3SOX** [comment]
@mmcnutt I think you can close this issue here now and follow the respective repos of @dhruv8sh

**Comment by Zren** [comment]
> Fixed! Since this issue does not seem to be watched over, I will publish it on the store myself.

I was waiting for a PR since there's [a dozen other widgets](https://github.com/Zren?tab=repositories&q=plasma-applet&type=&language=&sort=) to port and [I've only recently setup a Plasma 6 VM](https://zren.github.io/2024/03/04/opensuse-krypton-plasma-6-setup) and am currently working on a script to quickly replace keywords.

* https://github.com/Zren/plasma-applet-lib/commits/master/
* https://github.com/Zren/plasma-applet-lib/blob/master/kpac#L899

That said, I do have other widgets to port so you're free to maintain the widget going forward. I'll send you PRs if I add new features to the DesktopTheme python script.

> So I think it would be best if you could also publish the theme on the KDE Store with the control plasmoid as a dependency so it gets automatically installed, like it was before ^^

Ah, in the case that you do add new features and upload your own DesktopTheme, make sure to add ` [Plasma 6]` to the theme title, and I'll add ` [Plasma 5]` to mine.

**Comment by dhruv8sh** [comment]
Sounds great!
I am also working on a rust program to replace the keywords as well. Maybe we can collaborate on this.
